### PR TITLE
[FIX] mrp: open component quantity for kit

### PR DESCRIPTION
**Pull request by Whenrow**
The 'On Hand' button triggers action_open_quants on product model. It
do not explode the kit into its component

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/product.py [hunk 1]
```diff
@@ -502,8 +502,6 @@ def action_open_quants(self):
                 default_product_id=self.id,
                 single_product=True
             )
-        else:
-            self = self.with_context(product_tmpl_id=self.product_tmpl_id.id)
```
Comment (svs-odoo) [inline addons/stock/models/product.py:506]: Why did you remove this part ?

File: addons/mrp/models/product.py [hunk 2]
```diff
@@ -196,3 +196,18 @@ def action_view_mos(self):
...
+            if bom:
+                bom_kits[product] = bom
+        components = self - self.env['product.product'].concat(*list(bom_kits.keys()))
+        for product in bom_kits:
+            boms, bom_sub_lines = bom_kits[product].explode(product, 1)
+            components |= self.env['product.product'].concat(*[l[0].product_id for l in bom_sub_lines])
+        res = super(ProductProduct, components).action_open_quants()
+        res['context']['single_product'] = False
+        return res
+
```
Comment (svs-odoo) [inline addons/mrp/models/product.py:213]: A space too much :grimacing:

File: addons/stock/models/product.py (hunk 1 above)
Comment (Whenrow) [reply addons/stock/models/product.py:506]: `self.product_tmpl_id.id` doesn't work with differents products (from the kit) and I also saw it's not used in the action

File: addons/mrp/models/product.py [hunk 3]
```diff
@@ -196,3 +196,18 @@ def action_view_mos(self):
...
+        for product in self:
+            bom = self.env['mrp.bom']._bom_find(product=product, bom_type='phantom')
+            if bom:
+                bom_kits[product] = bom
+        components = self - self.env['product.product'].concat(*list(bom_kits.keys()))
+        for product in bom_kits:
+            boms, bom_sub_lines = bom_kits[product].explode(product, 1)
+            components |= self.env['product.product'].concat(*[l[0].product_id for l in bom_sub_lines])
+        res = super(ProductProduct, components).action_open_quants()
+        res['context']['single_product'] = False
```
Comment (svs-odoo) [inline addons/mrp/models/product.py:211]: With this line, `single_product` will always be false once MRP is installed.
In case this is a kit with multiple products, I think we shouldn't pass through the condition who set `single_product` to true.

File: addons/stock/models/stock_quant.py [hunk 4]
```diff
@@ -43,8 +43,8 @@ def _domain_product_id(self):
         if not self._is_inventory_mode():
             return
         domain = [('type', '=', 'product')]
-        if self.env.context.get('product_tmpl_id'):
-            domain = expression.AND([domain, [('product_tmpl_id', '=', self.env.context['product_tmpl_id'])]])
+        if self.env.context.get('product_tmpl_ids'):
+            domain = expression.AND([domain, [('product_tmpl_id', 'in', self.env.context['product_tmpl_ids'])]])
```
Comment (ryv-odoo) [inline addons/stock/models/stock_quant.py:47]: `self.env.context.get('product_tmpl_ids', []) + (self.env.context.get('product_tmpl_id') or [] )`

File: addons/stock/models/stock_quant.py [hunk 5]
```diff
@@ -43,8 +43,8 @@ def _domain_product_id(self):
...
         return domain
```
Comment (svs-odoo) [inline addons/stock/models/stock_quant.py:48]: I think you could had this condition to manage the case where your kit is constitute with only one product:
```suggestion
        if self.env.context.get('product_tmpl_ids'):
            domain = expression.AND([domain, [('product_tmpl_id', 'in', self.env.context['product_tmpl_ids'])]])
        elif self.env.context.get('default_product_id'):
            domain = expression.AND([domain, [('id', '=', self.env.context['default_product_id'])]])
        return domain
```
But as this case seems a little dubious, I don't really mind.

**Comment by Whenrow** [comment]
robodoo r+
